tests/test_tracking_cli.py

```diff
@@ -147,14 +156,19 @@ def test_experiments(wandb_mock, getargs_mock, caplog, samples_dir, tmp_dir):
     with (samples_dir / "experiments_wandb_calls.json").open("r") as f:
         assert log_calls == [call(**entry) for entry in json.load(f)]
     # Custom calls for .metrics files publication
-    assert [list(v.keys())[0] for c in metrics_calls for v in c.args] == [
+    expectations = [
```

Comment:
assert is a way to go in pytest, let's investigate what's wrong with it and fix it so that it works https://docs.pytest.org/en/7.1.x/how-to/assert.html

Reply:
alternatively you can disable the test and leave #365 for @vrigal  to investigate

Reply:
`assert` works just fine. The issue is that the log ordering is different in CI, it seems like there is some kind of race condition in the logging. It doesn't look like an error in the code, just an error in the test assertion. I raised an Exception in what I changed since it seemed to give more salient failures.

I'm fine changing it to a more idiomatic `assert` if that's your preference.

Reply:
yeah, I suggest switching to asserts since I think pytest might do some tricks there. If the order is not important you can use python sets instead of lists. If it is important, there's a bug.

Reply:
Ah ok, asserting on `set`s is much cleaner, thanks for the suggestion. I've updated the tests.